src/thor/triplegbuilder.cc

```diff
@@ -799,12 +813,35 @@ TripLeg_Edge* AddTripEdge(const AttributesController& controller,
 
   // Add names to edge if requested
   if (controller.attributes.at(kEdgeNames)) {
-    auto names_and_types = edgeinfo.GetNamesAndTypes();
+    std::vector<uint8_t> types;
+    auto names_and_types = edgeinfo.GetNamesAndTypes(types, true);
     trip_edge->mutable_name()->Reserve(names_and_types.size());
+    std::unordered_map<uint8_t, std::pair<uint8_t, std::string>> pronunciations =
+        edgeinfo.GetPronunciationsMap();
+    uint8_t name_index = 0;
     for (const auto& name_and_type : names_and_types) {
+      if (types.at(name_index) != 0) {
+        // Skip the tagged names
+        name_index++;
+        continue;
+      }
+
       auto* trip_edge_name = trip_edge->mutable_name()->Add();
+      // Assign name and type
       trip_edge_name->set_value(name_and_type.first);
       trip_edge_name->set_is_route_number(name_and_type.second);
+      std::unordered_map<uint8_t, std::pair<uint8_t, std::string>>::const_iterator iter =
+          pronunciations.find(name_index);
+
+      // Assign pronunciation alphabet and value if one exists
```

Comment:
Just like signs...a pronunciation has an index to a name...a name can have multiple pronunciations